cli/internal/reference/compref/compref.go

```diff
@@ -216,15 +261,13 @@ func Parse(input string) (*Ref, error) {
 	case *ctfv1.Repository:
 		t.Path = input
 	default:
-		return nil, fmt.Errorf("unsupported repository type: %q", ref.Type)
+		return nil, fmt.Errorf("unsupported repository type: %q", repoType)
```

Comment:
you have 2 type switch chases now, one here to set attributes, and one for type defaulting in https://github.com/open-component-model/open-component-model/pull/628/files#diff-db8ae1c2f0e634e09730aa65f488fa4e9ec05e7d8b295d17f706171e8ba908f9R22. Can they be unified?